main_test.go

```diff
@@ -217,4 +217,56 @@ func TestDownload_Chunks(t *testing.T) {
 	}
 }
 
-// TODO: add tests for getDownloadSize (success with Content-Length, success with Content-Range, failure)
+func TestGetDownloadSize_ContentLength(t *testing.T) {
+	s := httptest.NewServer(http.HandlerFunc(
+		func(w http.ResponseWriter, r *http.Request) {
+			fmt.Fprint(w, "Test")
+		},
+	))
+	defer s.Close()
+
+	d := DefaultDownloader()
+	d.client = &http.Client{Timeout: d.TimeoutPerChunk}
+	got, err := d.getDownloadSize(context.Background(), s.URL)
+
+	if err != nil {
+		t.Errorf("encountered an error:%s", err)
+	}
+	if got != 4 {
+		t.Errorf("invalid size, expected 4, got: %d", got)
+	}
+}
+
+func TestGetDownloadSize_ContentRange(t *testing.T) {
+	s := httptest.NewServer(http.HandlerFunc(
+		func(w http.ResponseWriter, r *http.Request) {
+			w.Header().Set("Content-Range", "bytes 1-10/123")
+			fmt.Fprint(w, "")
+		},
+	))
+	defer s.Close()
+
+	d := DefaultDownloader()
+	d.client = &http.Client{Timeout: d.TimeoutPerChunk}
+	got, err := d.getDownloadSize(context.Background(), s.URL)
+
+	if err != nil {
+		t.Errorf("encountered an error:%s", err)
+	}
+	if got != 123 {
+		t.Errorf("invalid size, expected 123, got: %d", got)
+	}
+}
+
+func TestGetDownloadSize_Failure(t *testing.T) {
+	d := DefaultDownloader()
+	d.client = &http.Client{Timeout: d.TimeoutPerChunk}
+	got, err := d.getDownloadSize(context.Background(), "test")
```

Comment:
Can we make this test more realistic (or add yet a fourth test case)?

Maybe a response with content-length set to 0 and no content-range header…?

Reply:
Yeah, good catch. However, not having Content-Range and ContentLength as 0 breaks the function, returning `18446744073709551615` and no error. Would you like me to add a sentinel to the function when no Content is presented?

Reply:
Ouch… another bug! 

Yes, if you can find a workaround, that would be great. Re-reading it, it looks like this logic is deceiving:

https://github.com/cuducos/chunk/blob/a27be85ef09709d8ac15e7804eedf09dbd879cd0/main.go#L150

Maybe we just check `resp.ContentLength <= 0` and deal with the other header inside of the `if` block to make it clearer and less error-prone.

Otherwise, feel free to leave the new test with the assertion commented with a `// TODO: fix` (or similar).

Reply:
Okay, so I encountered another bug... setting a sentinel for checking if both ContentLength and Content-Range are null breaks another test which expects an error in a different place... Is it okay if I remove the sentinel and leave this one for another issue?

Reply:
> Is it okay if I remove the sentinel and leave this one for another issue?

Sure think! Just leave the test you wrote commenting the `t.Errorf(…)` with a `// TODO: …` so we don't forget about it : )

Many many thanks 💜

Reply:
Plus, it might make more sense to return 0 instead of an error if ContentLegnth is 0 and there is no Content-Range header, and then checking if download size is 0. I will leave that for you to decide

Reply:
> it might make more sense to return 0 instead of an error if ContentLegnth is 0 

I don't think so. The only use case we have for getting the size is to know how many content-range requests to spin. If we don't know that, the whole process is broken.

In the future (stretch goal), we can capture cases like that and set a regular download (not using a content-range header), but for that, any error on `getDownloadSize` would do it.